packages/react-devtools-shared/src/backend/console.js

```diff
@@ -134,68 +152,78 @@ export function patch({
-                  // patch() is called by two places: (1) the hook and (2) the renderer backend.
-                  // The backend is what implements a message queue, so it's the only one that injects onErrorOrWarning.
-                  if (typeof onErrorOrWarning === 'function') {
-                    onErrorOrWarning(
-                      current,
-                      ((method: any): 'error' | 'warn'),
-                      // Copy args before we mutate them (e.g. adding the component stack)
-                      args.slice(),
-                    );
-                  }
-                }
+        let isInStrictMode = false;
 
-                if (shouldAppendWarningStack) {
-                  const componentStack = getStackByFiberInDevAndProd(
-                    workTagMap,
+        // Search for the first renderer that has a current Fiber.
+        // We don't handle the edge case of stacks for more than one (e.g. interleaved renderers?)
+        // eslint-disable-next-line no-for-of-loops/no-for-of-loops
+        for (const {
+          currentDispatcherRef,
+          getCurrentFiber,
+          onErrorOrWarning,
+          workTagMap,
+          getIsStrictMode,
+        } of injectedRenderers.values()) {
+          const current: ?Fiber = getCurrentFiber();
+          if (current != null) {
+            try {
+              if (typeof getIsStrictMode === 'function' && getIsStrictMode()) {
+                isInStrictMode = true;
+              }
+
+              if (shouldShowInlineWarningsAndErrors) {
+                // patch() is called by two places: (1) the hook and (2) the renderer backend.
+                // The backend is what implements a message queue, so it's the only one that injects onErrorOrWarning.
+                if (typeof onErrorOrWarning === 'function') {
+                  onErrorOrWarning(
                     current,
-                    currentDispatcherRef,
+                    ((method: any): 'error' | 'warn'),
+                    // Copy args before we mutate them (e.g. adding the component stack)
+                    args.slice(),
                   );
-                  if (componentStack !== '') {
-                    args.push(componentStack);
-                  }
                 }
-              } catch (error) {
-                // Don't let a DevTools or React internal error interfere with logging.
-              } finally {
-                break;
               }
+
+              if (shouldAppendWarningStack) {
+                const componentStack = getStackByFiberInDevAndProd(
+                  workTagMap,
+                  current,
+                  currentDispatcherRef,
+                );
+                if (componentStack !== '') {
+                  args.push(componentStack);
+                }
+              }
+            } catch (error) {
+              // Don't let a DevTools or React internal error interfere with logging.
```

Comment:
Should we rethrow in a timeout? This looks a bit risky since any mistake in the code would be forever suppressed and essentially invisible. Even if there isn’t one now, a mistake could be added later.